src/explore-education-statistics-admin/src/components/editable/EditableKeyStatDataBlock.tsx

```diff
@@ -0,0 +1,113 @@
+import Button from '@common/components/Button';
+import ButtonGroup from '@common/components/ButtonGroup';
+import LoadingSpinner from '@common/components/LoadingSpinner';
+import WarningMessage from '@common/components/WarningMessage';
+import useToggle from '@common/hooks/useToggle';
+import useKeyStatQuery from '@common/modules/find-statistics/hooks/useKeyStatQuery';
+import React from 'react';
+import EditableKeyStatDataBlockForm from '@admin/components/editable/EditableKeyStatDataBlockForm';
+import { KeyStatisticDataBlock } from '@common/services/publicationService';
+import EditableKeyStatDisplay from '@admin/components/editable/EditableKeyStatDisplay';
+
+interface KeyStatsDataFormValues {
+  trend: string;
+  guidanceTitle: string;
+  guidanceText: string;
+}
+
+export interface EditableKeyStatDataBlockProps {
+  keyStat: KeyStatisticDataBlock;
+  isEditing?: boolean;
+  isReordering?: boolean;
+  onRemove?: () => void;
+  onSubmit: (values: KeyStatsDataFormValues) => void;
+  testId?: string;
+}
+
+const EditableKeyStatDataBlock = ({
+  keyStat: {
+    id: keyStatId,
+    releaseId,
+    dataBlockId,
+    trend,
+    guidanceTitle = 'Help',
+    guidanceText,
+  },
+  isEditing = false,
+  isReordering = false,
+  testId = 'keyStat',
+  onRemove,
+  onSubmit,
+}: EditableKeyStatDataBlockProps) => {
+  const [showForm, toggleShowForm] = useToggle(false);
+  const [removing, toggleRemoving] = useToggle(false);
+
+  const { value: dataBlockValues, isLoading, error } = useKeyStatQuery(
+    releaseId,
+    dataBlockId,
+  );
+
+  if (isLoading) {
+    return <LoadingSpinner />;
```

Comment:
`LoadingSpinner` accepts a boolean `loading` param and accepts `children` so it's probably worth considering if `LoadingSpinner` can go elsewhere instead of it being in its own `if` statement

Reply:
Yeah, like:

```
return (
  <LoadingSpinner loading={isLoading}>
    ... the rest of the components here
  </LoadingSpinner>
);
```

but I can see also that there's a fair bit of complexity following this as to what exactly those children will be, so perhaps that's the reason that it's already this way?

Reply:
This was the way it was when I came to the code - I believe this file was original `EditablyeKeyStat.tsx`.

My preference would be to leave this as it is. If we switch, we're going to need to wrap all three other returns with `LoadingSpinner`.